batavia/VirtualMachine.js

```diff
@@ -1716,16 +1720,43 @@ VirtualMachine.prototype.byte_POP_EXCEPT = function() {
     this.unwind_block(block)
 }
 
-// VirtualMachine.prototype.byte_SETUP_WITH = function(dest) {
-//         ctxmgr = this.pop()
-//         this.push(ctxmgr.__exit__)
-//         ctxmgr_obj = ctxmgr.__enter__()
-//         if PY2:
-//             this.push_block('with', dest)
-//         elif PY3:
-//             this.push_block('finally', dest)
-//         this.push(ctxmgr_obj)
-// }
+VirtualMachine.prototype.byte_SETUP_WITH = function(dest) {
+    var mgr = this.top()
```

Comment:
CPython would pop context manager from the stack here and push only `__exit__` method of the context manager to call it  as a function in `WITH_CLEANUP`. However I couldn't make it work this way, so I am leaving context manager on the stack and in `WITH_CLEANUP` I am calling `mgr.__exit__` (line 1753). I don't see how this would break anything. What do you think @freakboy3742 ?